openwisp_controller/config/sortedm2m/fields.py

```diff
@@ -0,0 +1,52 @@
+"""
+Customized sortedm2m field to send 'pre_add' signal on all occurrences.
+"""
+
+from django.db import router
+from django.db.models import signals
+from django.utils.functional import cached_property
+from sortedm2m.fields import (
+    SortedManyToManyDescriptor as BaseSortedManyToManyDescriptor,
+)
+from sortedm2m.fields import SortedManyToManyField as BaseSortedManyToManyField
+from sortedm2m.fields import (
+    create_sorted_many_related_manager as base_create_sorted_many_related_manager,
+)
+
+
+def create_sorted_many_related_manager(superclass, rel, *args, **kwargs):
+    BaseSortedRelatedManager = base_create_sorted_many_related_manager(
+        superclass, rel, *args, **kwargs
+    )
+
+    class SortedRelatedManager(BaseSortedRelatedManager):
+        def _add_items(self, source_field_name, target_field_name, *objs, **kwargs):
+            db = router.db_for_write(self.through, instance=self.instance)
+            super()._add_items(source_field_name, target_field_name, *objs, **kwargs)
+            if not objs:
+                signals.m2m_changed.send(
+                    sender=self.through,
+                    action='custom_openwisp',
+                    instance=self.instance,
+                    reverse=self.reverse,
+                    model=self.model,
+                    pk_set=set(),
+                    using=db,
+                )
```

Comment:
Here I'd send both pre_add and post_add.
This will should automatically fix https://github.com/openwisp/openwisp-controller/issues/438 as well, which is related to this bug (always a sortedm2m issue), can you write a test for it in a separate commit please? I expect the test to pass once we send pre_add and post_add also if `objs` is empty.

Reply:
I have added a test for this in 262a7f9

But as you mentioned in the issue description. The configuration status changes to `modified`on saving the device even when no template is selected. 

> There's on caveat: we must ensure that saving a device which does not have any templates repeatedly does not set the status as modified.